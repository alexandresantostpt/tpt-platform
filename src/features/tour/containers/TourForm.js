import { connect } from 'react-redux'

import { getObjSelector as travelGetObjSelector, getClientsSelector } from '@features/travelScript/selectors'

import { getObjSelector } from '../selectors'
import { requestDelAction, requestGetSingleAction, requestSaveDataAction } from '../actions'

import TourForm from '../pages/TourForm'

import { resetSuggestionsAction, searchLibraryAction } from '@features/library/actions'
import { getListSelector } from '@features/library/selectors'

const mapStateToProps = state => ({
    clients: getClientsSelector(state),
    tour: getObjSelector(state),
    tourSuggestions: getListSelector(state),
    travelScript: travelGetObjSelector(state)
})

const mapDispatchToProps = {
    requestDelAction,
    requestGetSingleAction,
    requestSaveDataAction,
    resetSuggestionsAction,
    searchLibraryAction
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(TourForm)
