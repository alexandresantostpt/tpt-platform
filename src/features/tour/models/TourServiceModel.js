import { config } from '@config'

import moment from 'moment'

import { normalizeTimezone } from '@utils/date'

import TourLibraryModel from './TourLibraryModel'

class TourServiceModel {
    constructor(
        {
            __v,
            _id,
            address,
            createdAt,
            guideOrDriver,
            library,
            localOperator,
            observation,
            outDate,
            outHour,
            passengers,
            peopleCount,
            scriptDate,
            transferType,
            updatedAt
        },
        fromForm = false
    ) {
        this.__v = __v
        this._id = _id
        this._address = address
        this._createdAt = moment(createdAt).utc(false)
        this._guideOrDriver = guideOrDriver
        this._library = library ? new TourLibraryModel(library) : null
        this._localOperator = localOperator
        this._observation = observation
        this._outDate = moment(outDate).utc(false)
        this._outHour = fromForm ? normalizeTimezone(outHour) : moment(outHour).utc(false)
        this._passengers = passengers
        this._peopleCount = peopleCount
        this._scriptDate = scriptDate
        this._transferType = transferType
        this._updatedAt = moment(updatedAt).utc(false)
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get address() {
        return this._address
    }

    get createdAt() {
        return this._createdAt
    }

    get guideOrDriver() {
        return this._guideOrDriver
    }

    get library() {
        return this._library
    }

    get localOperator() {
        return this._localOperator
    }

    get observation() {
        return this._observation
    }

    get outDate() {
        return this._outDate
    }

    get outHour() {
        return this._outHour
    }

    get passengers() {
        return this._passengers
    }

    get peopleCount() {
        return this._peopleCount
    }

    get scriptDate() {
        return this._scriptDate
    }

    get transferType() {
        return this._transferType
    }

    get updatedAt() {
        return this._updatedAt
    }

    getImage() {
        return this.library && this.library.image && `${config.api.url}/tour/download/${this.library.image}`
    }

    /* eslint-disable-next-line complexity */
    toDTO() {
        return {
            __v: this.v,
            _id: this.id,
            address: this.address || null,
            guideOrDriver: this.guideOrDriver || null,
            library: this.library ? this.library.toDTO() : null,
            localOperator: this.localOperator || null,
            observation: this.observation || null,
            outDate: this.outDate || null,
            outHour: this.outHour || null,
            passengers: this.passengers || null,
            peopleCount: this.peopleCount || null,
            scriptDate: this.scriptDate || null,
            transferType: this.transferType || null
        }
    }

    toForm() {
        return {
            __v: this.__v,
            _id: this.id,
            address: this.address,
            guideOrDriver: this.guideOrDriver,
            library: this.library ? this.library.toForm() : null,
            localOperator: this.localOperator,
            observation: this.observation,
            outDate: this.outDate,
            outHour: this.outHour,
            passengers: this.passengers,
            peopleCount: this.peopleCount,
            script: this.script,
            scriptDate: this.scriptDate,
            transferType: this.transferType
        }
    }
}

export default TourServiceModel
