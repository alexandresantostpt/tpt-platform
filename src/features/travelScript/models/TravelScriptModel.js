import moment from 'moment'

import Travel from '../../travel/TravelModel'
import TravelDateModel from './TravelDateModel'
import { isNotEmpty } from '@/utils/functions'

class TravelScriptModel {
    constructor({ __v, _id, dates, deleted, edited, travel, createdAt, updatedAt }) {
        this.__v = __v
        this._id = _id
        this._dates = dates && [...dates].map(date => new TravelDateModel(date))
        this._deleted = deleted
        this._edited = edited
        this._travel = new Travel(travel)
        this._createdAt = moment(createdAt).utc(false)
        this._updatedAt = moment(updatedAt).utc(false)
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get dates() {
        return this._dates
    }

    get deleted() {
        return this._deleted
    }

    get edited() {
        return this._edited
    }

    get travel() {
        return this._travel
    }

    get createdAt() {
        return this._createdAt
    }

    get updatedAt() {
        return this._updatedAt
    }

    hasServices() {
        return this.dates.some(date => date.services.length)
    }

    getRangeDates() {
        if (isNotEmpty(this._travel)) {
            return { endDate: this._travel.travelEndDate, startDate: this._travel.travelStartDate }
        }

        return {}
    }

    getClientsId() {
        return [...this.travel.clients].map(({ _id }) => _id)
    }

    toForm() {
        return {
            dates: this.dates,
            edited: this.edited,
            id: this.id,
            travel: this.travel,
            v: this.v
        }
    }
}

export default TravelScriptModel
