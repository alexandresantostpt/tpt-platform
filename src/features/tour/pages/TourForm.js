import { config } from '@config'
import { events } from '@constants/events'

import PubSub from 'pubsub-js'
import PropTypes from 'prop-types'
import React, { useEffect, useRef, useState } from 'react'

import { colors } from '@helpers/colors'
import { getQueryParams } from '@utils/browser'
import { i18n } from '@i18n'
import { isNotUndefined } from '@utils/functions'
import { services, librariesServices } from '@/constants/services'
import { withRouter } from 'react-router-dom'

import Button from '@components/button/Button'
import Cancel from '@components/cancel/Cancel'
import FormBuilder from '@components/form/FormBuilder'
import Panel from '@components/panel/Panel'

import FormButtonContainer from '@components/form/container/button/FormButtonContainer'
import WrapperPadding from '@components/wrapper/WrapperPadding'

import { tourFormFields } from '../fields'

import TourServiceModel from '../models/TourServiceModel'

const TourForm = props => {
    const { match, resetSuggestionsAction, searchLibraryAction, tour, tourSuggestions, travelScript } = props
    const {
        params: { id, idTour }
    } = match
    const { cityId, dateId, scriptDate, scriptId } = getQueryParams()

    const form = useRef()
    const previousPath = `/travel/${id}/script`

    const [image, updateImage] = useState()
    const [serviceDisabled, updateServiceDisabled] = useState(false)
    const [newImage, updateNewImage] = useState(null)

    useEffect(() => {
        PubSub.subscribe(events.UPDATE_IMAGE, (_, libraryImage) => {
            if (libraryImage) {
                updateNewImage(`${config.api.url}/tour/download/${libraryImage}`)
            }
        })
        return () => PubSub.unsubscribe(events.UPDATE_IMAGE)
    }, [])

    useEffect(() => {
        const { requestGetSingleAction } = props
        if (idTour) {
            requestGetSingleAction({ id: idTour })
            updateServiceDisabled(true)
        } else {
            requestGetSingleAction({ id: null })
            updateServiceDisabled(false)
        }
    }, [idTour])

    useEffect(() => {
        resetSuggestionsAction()
    }, [scriptDate])

    const callSaveData = values => {
        const { requestSaveDataAction } = props
        requestSaveDataAction({
            data: {
                dateId,
                scriptId,
                service: new TourServiceModel({ ...values, scriptDate }, true).toDTO(),
                type: services.TOUR
            },
            image,
            redirectTo: previousPath
        })
        reset()
    }

    const callSearchLibrary = name => {
        if (isNotUndefined(cityId)) {
            searchLibraryAction({
                cityId: cityId,
                libraryName: name,
                type: librariesServices.TOUR,
                updateRecords: true
            })
        }
    }

    const onSaveClick = () => form.current.form.submit()

    const reset = () => {
        PubSub.publish(events.FORM_RESET)
        updateImage(null)
    }

    return (
        <>
            <WrapperPadding>
                <Panel backgroundImageUrl={tour ? tour.getImage() : newImage || null} icon="tour" uploadImage={updateImage} />
            </WrapperPadding>
            <FormBuilder
                clients={props.clients}
                disabled={serviceDisabled}
                fields={tourFormFields}
                getSuggestions={callSearchLibrary}
                initialValues={tour ? tour.toForm() : { outDate: scriptDate }}
                onSubmit={callSaveData}
                rangeDates={travelScript && travelScript.getRangeDates()}
                ref={process.env.__TEST__ ? null : form}
                resetSuggestions={resetSuggestionsAction}
                suggestions={tourSuggestions}
            />
            <FormButtonContainer>
                <Button
                    color={colors.purple}
                    onClick={() => (serviceDisabled ? updateServiceDisabled(false) : onSaveClick())}
                    text={i18n.t(serviceDisabled ? 'buttons.editService' : 'buttons.saveService')}
                />
                <Cancel link={previousPath} />
            </FormButtonContainer>
        </>
    )
}

TourForm.defaultProps = {
    clients: [],
    tour: {},
    tourSuggestions: [],
    travelScript: {}
}

TourForm.propTypes = {
    clients: PropTypes.array,
    match: PropTypes.object.isRequired,
    requestDelAction: PropTypes.func.isRequired,
    requestSaveDataAction: PropTypes.func.isRequired,
    resetSuggestionsAction: PropTypes.func.isRequired,
    searchLibraryAction: PropTypes.func.isRequired,
    tour: PropTypes.objectOf(TourServiceModel),
    tourSuggestions: PropTypes.array,
    travelScript: PropTypes.object
}

export default withRouter(TourForm)
