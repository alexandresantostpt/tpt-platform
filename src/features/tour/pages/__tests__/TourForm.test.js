import React from 'react'
import renderer from 'react-test-renderer'

import TourServiceModel from '../../models/TourServiceModel'
import TourForm from '../TourForm'
import TravelScriptModel from '@/features/travelScript/models/TravelScriptModel'

describe('Tests for TourForm component', () => {
    const tour = new TourServiceModel({
        __v: 0,
        _id: '5c825ffc650a395fa6d49254',
        createdAt: '2019-03-08T12:28:44.199Z',
        describe: 'Teste',
        duration: '10 minutos',
        guideOrDriver: 'Guia',
        includedMeals: [],
        language: 'PortuguÃªs',
        localOperator: 'Teste',
        outDate: '2019-02-01T02:00:00.000Z',
        outHour: '2019-02-01T02:00:00.000Z',
        passenger: ['5c757fb2f1a4a01bbfe7b21b'],
        peopleCount: 5,
        script: '5c757fb2f1a4a01bbfe7b21d',
        scriptDate: '2019-02-04T00:00:00.000Z',
        title: 'Teste',
        tourType: 'Privativo',
        transferType: 'teste',
        type: 'TOUR',
        updatedAt: '2019-03-08T12:28:44.199Z'
    })
    const travelScript = new TravelScriptModel({ travel: { __v: 0 } })

    it('Should render withou crash', () => {
        const match = {
            params: {
                id: '5c52ebcd5fc7c949223ab705'
            }
        }
        const tree = renderer
            .create(
                <TourForm
                    match={match}
                    requestDelAction={jest.fn()}
                    requestGetSingleAction={jest.fn()}
                    requestSaveDataAction={jest.fn()}
                    resetSuggestionsAction={jest.fn()}
                    tour={tour}
                    travelScript={travelScript}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
