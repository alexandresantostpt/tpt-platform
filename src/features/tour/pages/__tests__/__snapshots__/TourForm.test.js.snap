// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for TourForm component Should render withou crash 1`] = `
Array [
  <div
    className="WrapperPadding-ovjbyx-0 lcnoja"
  >
    <div
      className="Panel__StyledPanel-sc-1qzkpb6-0 cGUlCn"
    >
      <span
        className="Panel__StyledIcon-sc-1qzkpb6-2 iavvSk"
      >
        <span
          className="icon icon-tour Icon__StyledIcon-crr1cg-0 dfhcZR"
          disabled={false}
          fontSize="3rem"
          onClick={null}
        />
      </span>
      <div
        className="UploadButton__StyledContainer-sc-1p8zoc-0 fpZFIG"
      >
        <label
          className="UploadButton__StyledUploadButton-sc-1p8zoc-3 ivhrYW"
          htmlFor="inputFile"
        >
          +
        </label>
        <input
          accept="image/png, image/jpeg"
          className="UploadButton__InputFile-sc-1p8zoc-4 eLaooA"
          id="inputFile"
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
    <ul
      className="PanelActions__StyledActions-sc-8xk8c8-0 lllkGw"
    />
  </div>,
  <form
    className="FormBuilder__StyledForm-c1htou-0 PZxQa"
    onSubmit={[Function]}
  >
    <div
      className="form__container"
    >
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Título do passeio
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormSuggest__StyledAutosuggest-d8stcq-0 gUlkuf"
        >
          <div
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-library.name"
            className="react-autosuggest__container"
            role="combobox"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls="react-autowhatever-library.name"
              autoComplete="off"
              className="react-autosuggest__input"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
            <div
              className="react-autosuggest__suggestions-container"
              id="react-autowhatever-library.name"
              role="listbox"
            />
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Itinerário/descrição do passeio
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <textarea
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormTextArea__StyledFormText-sc-9g3u2c-0 hmDUjT"
          data-cy="input-library.describe"
          disabled={false}
          id="describe"
          isInvalid={false}
          name="library.describe"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Data do passeio
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormDate__StyledFormDate-sc-1cgz0rk-0 ebKpCI"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-"
                    aria-label="Selecione uma data..."
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                    disabled={false}
                    id=""
                    name=""
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Selecione uma data..."
                    readOnly={true}
                    required={false}
                    type="text"
                    value="01 Feb 2019"
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon icon-calendar Icon__StyledIcon-crr1cg-0 cmjVJQ"
                    disabled={false}
                    fontSize="1.75rem"
                    onClick={null}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Horário de saída
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="02:00"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Endereço
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-address"
          disabled={false}
          id="address"
          maxLength={255}
          name="address"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Telefone de contato
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.phoneNumber"
          disabled={false}
          id="phoneNumber"
          maxLength={30}
          name="library.phoneNumber"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Site
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-site"
          disabled={false}
          id="site"
          maxLength={255}
          name="site"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Duração aproximada do passeio
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.duration"
          disabled={false}
          id="duration"
          maxLength={15}
          name="library.duration"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Quantidade de viajantes
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-peopleCount"
          disabled={false}
          id="peopleCount"
          maxLength={255}
          min={1}
          name="peopleCount"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="number"
          value={5}
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Operador local
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-localOperator"
          disabled={false}
          id="localOperator"
          maxLength={120}
          name="localOperator"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Teste"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Meio de transporte
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-transferType"
          disabled={false}
          id="transferType"
          maxLength={30}
          name="transferType"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="teste"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Linguagem
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.language"
          disabled={false}
          id="language"
          maxLength={30}
          name="library.language"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Tipo de passeio
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormRadio__StyledOption-xdzjtw-1 dZGJdF"
        >
          <input
            checked={false}
            disabled={false}
            id="library.tourType-Privativo"
            name="library.tourType-Privativo"
            onChange={[Function]}
            type="radio"
            value="Privativo"
          />
          <span
            className="FormRadio__StyledLabel-xdzjtw-0 cnAFWF"
          >
            Privativo
          </span>
        </div>
        <div
          className="FormRadio__StyledOption-xdzjtw-1 dZGJdF"
        >
          <input
            checked={false}
            disabled={false}
            id="library.tourType-Regular"
            name="library.tourType-Regular"
            onChange={[Function]}
            type="radio"
            value="Regular"
          />
          <span
            className="FormRadio__StyledLabel-xdzjtw-0 cnAFWF"
          >
            Regular
          </span>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Guia ou motorista
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormRadio__StyledOption-xdzjtw-1 dZGJdF"
        >
          <input
            checked={true}
            disabled={false}
            id="guideOrDriver-Guia"
            name="guideOrDriver-Guia"
            onChange={[Function]}
            type="radio"
            value="Guia"
          />
          <span
            className="FormRadio__StyledLabel-xdzjtw-0 cnAFWF"
          >
            Guia
          </span>
        </div>
        <div
          className="FormRadio__StyledOption-xdzjtw-1 dZGJdF"
        >
          <input
            checked={false}
            disabled={false}
            id="guideOrDriver-Motorista"
            name="guideOrDriver-Motorista"
            onChange={[Function]}
            type="radio"
            value="Motorista"
          />
          <span
            className="FormRadio__StyledLabel-xdzjtw-0 cnAFWF"
          >
            Motorista
          </span>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Refeições inclusas
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormCheck__StyledOption-sc-4ngozm-1 Tgnbj"
        >
          <input
            checked={false}
            id="breakfast"
            name="breakfast"
            onChange={[Function]}
            type="checkbox"
            value="breakfast"
          />
          <span
            className="FormCheck__StyledLabel-sc-4ngozm-0 ijqNlP"
          >
            Café da manhã
          </span>
        </div>
        <div
          className="FormCheck__StyledOption-sc-4ngozm-1 Tgnbj"
        >
          <input
            checked={false}
            id="lunch"
            name="lunch"
            onChange={[Function]}
            type="checkbox"
            value="lunch"
          />
          <span
            className="FormCheck__StyledLabel-sc-4ngozm-0 ijqNlP"
          >
            Almoço
          </span>
        </div>
        <div
          className="FormCheck__StyledOption-sc-4ngozm-1 Tgnbj"
        >
          <input
            checked={false}
            id="dinner"
            name="dinner"
            onChange={[Function]}
            type="checkbox"
            value="dinner"
          />
          <span
            className="FormCheck__StyledLabel-sc-4ngozm-0 ijqNlP"
          >
            Jantar
          </span>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Selecionar passageiros
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Observações
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-observation"
          disabled={false}
          id="observation"
          maxLength={255}
          name="observation"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
  </form>,
  <div
    className="FormButtonContainer-sc-1cqqivn-0 ADyLt"
  >
    <button
      className="Button__StyledButton-sc-1fc3klx-0 bEGBgE"
      disabled={false}
      fill={false}
      onClick={[Function]}
    >
      Salvar serviço
    </button>
    <a
      className="Cancel__StyledCancel-sc-1prgtit-0 WzuoR"
      to="/travel/5c52ebcd5fc7c949223ab705/script"
    >
      Cancelar
    </a>
  </div>,
]
`;
